File: 2024/lua/aoc.lua
local function read_file(path)
    local file = io.open(path, "r")
    if not file then
        error("Could not open file " .. path)
    end
    local content = file:read("*all")
    file:close()
    return content
end

function Run(path, solution)
    io.write("@=== " .. path .. "\n| ")
    local start_time = os.clock()
    local file = read_file(path)
    solution(file)
    local elapsed = os.clock() - start_time;
    print("| time: " .. (elapsed * 1000) .. "ms");
end

File: 2024/js/aoc.js
import fs from 'fs';

function readInput(path) {
    return fs.readFileSync(path, 'utf8');
}

export function run(path, solution) {
    console.log(`==== ${path}`);
    const startTime = process.hrtime();
    solution(readInput(path));
    "\x1b[2J"
    const elapsedTime = process.hrtime(startTime);
    console.log(`==== time: ${elapsedTime[0]}s ${elapsedTime[1] / 1e6}ms`);
}
